app.py

```diff
@@ -53,11 +55,40 @@ def inject_configurations():
 # Routes
 @app.route('/')
 @login_required
-def index():  # put application's code here
+def index():
     current_year = get_current_year()
+    year = db.session.query(Year).filter_by(year=current_year).first()
     user = db.session.query(User).filter_by(email=session['email']).first()
-    courses_teacher = db.session.query(Course).filter_by(year=current_year).join(Teacher).filter(Teacher.user_id == user.id).all()
-    return render_template("home.html", user=user, courses=courses_teacher, evaluations=user.evaluations)
+    researcher = db.session.query(Researcher).filter_by(user_id=user.id).first()
+    teacher = db.session.query(Teacher).filter_by(user_id=user.id).first() if user.is_teacher else None
+
+    # Get the teacher's courses
+    courses_teacher = db.session.query(Course).join(Teacher).filter(
+        and_(Teacher.user_id == user.id, Course.year == current_year)
+    ).all() if year.publication_status != 'Draft' and user.is_teacher else []
+
```

Comment:
Why check if the user is a teacher ? If they are not your join between the Teacher and Course tables should be empty. So there is no need to make that check.

And why do a check if the publication status from the current_year is not a draft. As mentioned in another comment, having the publication_status in the Year table is irrelevant. Are you trying to filter to only display the courses that have been assigned to researchers ? I don't think you need it, but if you really want that you need to check it through the `assignement_published` table.

Because for the moment, courses that have not been assigned to researchers can't be viewed by the teachers.